sagemaker-featurestore/feature_store_securely_store_images.ipynb

```diff
@@ -0,0 +1,680 @@
+    "\n",
+    "print('Running ' + query_table)\n",
+    "\n",
+    "# Run the Athena query\n",
+    "query.run(query_string=query_table, output_location='s3://'+s3_bucket_name+'/'+prefix+'/query_results/')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "time.sleep(60)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset = query.as_dataframe()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(dataset.dtypes)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Below is the data queried from your feature store."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def parse_show_image(df):\n",
+    "    \"\"\"\n",
+    "    df: dataframe of your data\n",
+    "    \n",
+    "    Return a numpy array of your images that have been reshaped into it's corresponding shape. \n",
+    "    \"\"\"\n",
+    "    import ast\n",
+    "    images = []\n",
+    "    for index, entry in enumerate(np.array(df['img'])):\n",
+    "        entry = entry.strip('b')\n",
+    "        entry = np.array(ast.literal_eval(ast.literal_eval(entry)))\n",
+    "        shape = ast.literal_eval(df['shape'][index])\n",
+    "        entry = entry.reshape(shape[0], shape[1])\n",
+    "        images.append(entry)\n",
+    "    return np.array(images)\n",
+    "images = parse_show_image(dataset)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Below shows the shape of your image data set. \n",
+    "images.shape"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Plot the images to demonstrate that you can view the images stored in your feature store. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for img in images:\n",
+    "    plt.imshow(img)\n",
+    "    plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Clean Up Resources\n",
+    "Remove the Feature Group that was created. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "feature_group.delete()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Next Steps\n",
+    "\n",
+    "In this notebook we covered how to securely store data sets of images in a feature store using KMS key.\n",
```

Comment:
"In this notebook, we covered how to securely store datasets of images in a feature store using a KMS key.\n",